diff --git a/Source/monster.cpp b/Source/monster.cpp
--- a/Source/monster.cpp
+++ b/Source/monster.cpp
@@ -4470,3 +4470,34 @@
+void PrintMonsterResistances(const Monster &monster)
+{
+	const int res = monster.resistance & (RESIST_MAGIC | RESIST_FIRE | RESIST_LIGHTNING | IMMUNE_MAGIC | IMMUNE_FIRE | IMMUNE_LIGHTNING);
+
+	if ((res & (RESIST_MAGIC | RESIST_FIRE | RESIST_LIGHTNING)) != 0) {
+		std::string resists = std::string(_("Resists:"));
+		if ((res & RESIST_MAGIC) != 0)
+			resists.append(_(" Magic"));
+		if ((res & RESIST_FIRE) != 0)
+			resists.append(_(" Fire"));
+		if ((res & RESIST_LIGHTNING) != 0)
+			resists.append(_(" Lightning"));
+		AddInfoBoxString(resists);
+	} else {
+		AddInfoBoxString(_("No resistances"));
+	}
+
+	if ((res & (IMMUNE_MAGIC | IMMUNE_FIRE | IMMUNE_LIGHTNING)) != 0) {
+		std::string immune = std::string(_("Immune:"));
+		if ((res & IMMUNE_MAGIC) != 0)
+			immune.append(_(" Magic"));
+		if ((res & IMMUNE_FIRE) != 0)
+			immune.append(_(" Fire"));
+		if ((res & IMMUNE_LIGHTNING) != 0)
+			immune.append(_(" Lightning"));
+		AddInfoBoxString(immune);
+	} else {
+		AddInfoBoxString(_("No Immunities"));
+	}
+}
+
 void PrintMonstHistory(int mt)
 {
 	if (*GetOptions().Gameplay.showMonsterType) {
@@ -4500,33 +4531,8 @@
 		}
 		AddInfoBoxString(FormatRuntime(_("Hit Points: {:d}-{:d}"), minHP, maxHP));
 	}
-	if (MonsterKillCounts[mt] >= 15) {
-		const int res = (sgGameInitInfo.nDifficulty != DIFF_HELL) ? MonstersData[mt].resistance : MonstersData[mt].resistanceHell;
-		if ((res & (RESIST_MAGIC | RESIST_FIRE | RESIST_LIGHTNING | IMMUNE_MAGIC | IMMUNE_FIRE | IMMUNE_LIGHTNING)) == 0) {
-			AddInfoBoxString(_("No magic resistance"));
-		} else {
-			if ((res & (RESIST_MAGIC | RESIST_FIRE | RESIST_LIGHTNING)) != 0) {
-				std::string resists = std::string(_("Resists:"));
-				if ((res & RESIST_MAGIC) != 0)
-					resists.append(_(" Magic"));
-				if ((res & RESIST_FIRE) != 0)
-					resists.append(_(" Fire"));
-				if ((res & RESIST_LIGHTNING) != 0)
-					resists.append(_(" Lightning"));
-				AddInfoBoxString(resists);
-			}
-			if ((res & (IMMUNE_MAGIC | IMMUNE_FIRE | IMMUNE_LIGHTNING)) != 0) {
-				std::string immune = std::string(_("Immune:"));
-				if ((res & IMMUNE_MAGIC) != 0)
-					immune.append(_(" Magic"));
-				if ((res & IMMUNE_FIRE) != 0)
-					immune.append(_(" Fire"));
-				if ((res & IMMUNE_LIGHTNING) != 0)
-					immune.append(_(" Lightning"));
-				AddInfoBoxString(immune);
-			}
-		}
-	}
+
+	PrintMonsterResistances(Monsters[pcursmonst]);
 }
 
 void PrintUniqueHistory()
@@ -4536,19 +4542,5 @@
 		AddInfoBoxString(FormatRuntime(_("Type: {:s}"), GetMonsterTypeText(monster.data())));
 	}
 
-	const int res = monster.resistance & (RESIST_MAGIC | RESIST_FIRE | RESIST_LIGHTNING | IMMUNE_MAGIC | IMMUNE_FIRE | IMMUNE_LIGHTNING);
-	if (res == 0) {
-		AddInfoBoxString(_("No resistances"));
-		AddInfoBoxString(_("No Immunities"));
-	} else {
-		if ((res & (RESIST_MAGIC | RESIST_FIRE | RESIST_LIGHTNING)) != 0)
-			AddInfoBoxString(_("Some Magic Resistances"));
-		else
-			AddInfoBoxString(_("No resistances"));
-		if ((res & (IMMUNE_MAGIC | IMMUNE_FIRE | IMMUNE_LIGHTNING)) != 0) {
-			AddInfoBoxString(_("Some Magic Immunities"));
-		} else {
-			AddInfoBoxString(_("No Immunities"));
-		}
-	}
+	PrintMonsterResistances(monster);
 }
