diff --git a/Source/objects.h b/Source/objects.h
--- a/Source/objects.h
+++ b/Source/objects.h
@@ -284,7 +284,7 @@ struct Object {
 extern DVL_API_FOR_TEST Object Objects[MAXOBJECTS];
 extern int AvailableObjects[MAXOBJECTS];
 extern int ActiveObjects[MAXOBJECTS];
-extern int ActiveObjectCount;
+extern DVL_API_FOR_TEST int ActiveObjectCount;
 /** @brief Indicates that objects are being loaded during gameplay and pre calculated data should be updated. */
 extern bool LoadingMapObjects;
 /** Tracks progress through the tome sequence that spawns Na-Krul (see OperateNakrulBook()) */
